compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -739,7 +747,12 @@ fn transform_ty<'tcx>(tcx: TyCtxt<'tcx>, ty: Ty<'tcx>, options: TransformTyOptio
     let mut ty = ty;
 
     match ty.kind() {
-        ty::Float(..) | ty::Char | ty::Str | ty::Never | ty::Foreign(..) => {}
+        ty::Float(..)
+        | ty::Char
+        | ty::Str
+        | ty::Never
+        | ty::Foreign(..)
+        | ty::GeneratorWitness(..) => {}
```

Comment:
Why aren't you transforming any of the types *within* the generator?

Reply:
Neither the generator types nor their substs need to be transformed, just encoded. `transform_ty` coalesces types that have the same encoding into a single entity (e.g., c_void into unit), normalizes, and generalizes types before handing them to encode_ty. It greatly simplifies the encoding and compression/substitution logic.

Reply:
What if a generator captures a `c_void` across an await point? Why are all the other transformations in this fn happening structurally except for generators? Still not sure if I understand.

Reply:
Ahh, indeed. You're right. I should traverse the ty::GeneratorWitness substs looking for types that need to be transformed--I'll fix it. Thanks for pointing that out!

Reply:
Done.